cider/tests/benchmark/CiderBenchmarkUtils.h

```diff
@@ -0,0 +1,111 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <arrow/api.h>
+#include <arrow/compute/kernel.h>
+#include <arrow/csv/api.h>
+#include <arrow/io/api.h>
+#include <arrow/io/file.h>
+#include <arrow/stl.h>
+#include <arrow/util/iterator.h>
+#include "exec/plan/parser/TypeUtils.h"
+#include "tests/utils/ArrowArrayBuilder.h"
+#include "tests/utils/Utils.h"
+
+void constructBuilder(substrait::Type s_type,
+                      ArrowArrayBuilder& builder,
+                      std::shared_ptr<arrow::ChunkedArray> field,
+                      const std::string& name) {
+  for (int j = 0; j < field->num_chunks(); j++) {
+    auto chunk = field->chunk(j);
+    builder.addColumn(
+        name, s_type, chunk->null_bitmap_data(), chunk->data()->buffers[1]->data());
+  }
+}
+
+void constructStringBuilder(substrait::Type s_type,
+                            ArrowArrayBuilder& builder,
+                            std::shared_ptr<arrow::ChunkedArray> field,
+                            const std::string& name) {
+  for (int j = 0; j < field->num_chunks(); j++) {
+    auto chunk = field->chunk(j);
+    builder.addColumn(name,
+                      s_type,
+                      chunk->null_bitmap_data(),
+                      chunk->data()->buffers[1]->data(),
+                      chunk->data()->buffers[2]->data());
+  }
+}
+
+void convertToCiderBatch(ArrowArrayBuilder& builder,
+                         std::shared_ptr<arrow::ChunkedArray> field,
+                         const std::string& name) {
+  auto tag = field->type()->id();
+  switch (tag) {
+    case arrow::Type::type::INT32:
+      return constructBuilder(CREATE_SUBSTRAIT_TYPE(I32), builder, field, name);
+    case arrow::Type::type::INT64:
+      return constructBuilder(CREATE_SUBSTRAIT_TYPE(I64), builder, field, name);
+    case arrow::Type::type::FLOAT:
+      return constructBuilder(CREATE_SUBSTRAIT_TYPE(Fp32), builder, field, name);
+    case arrow::Type::type::DOUBLE:
+      return constructBuilder(CREATE_SUBSTRAIT_TYPE(Fp64), builder, field, name);
+    case arrow::Type::type::STRING:
+      return constructStringBuilder(CREATE_SUBSTRAIT_TYPE(Varchar), builder, field, name);
+  }
+}
+
+std::shared_ptr<CiderBatch> readFromCsv(std::string csv_file,
```

Comment:
I suggest to refactor this part to provide a reader base class, and impl a derived CsvReader. maybe we need ParquetReader in the future. reader.readAll should return ArrowArray/ArrowSchema